android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -0,0 +1,251 @@
+    private AdvertiseData getAdvertisementData(String uri) {
+        AdvertiseData.Builder builder = new AdvertiseData.Builder();
+        builder.setIncludeTxPowerLevel(false); // reserve advertising space for URI
+
+        // Manually build the advertising info
+        // See https://github.com/google/eddystone/tree/master/eddystone-url
+        byte[] urlData = AdvertiseDataUtils.encodeUri(uri);
+        if (urlData == null || urlData.length == 0) {
+            return null;
+        }
+
+        byte[] beaconData = new byte[urlData.length + 2];
+        System.arraycopy(urlData, 0, beaconData, 2, urlData.length);
+        beaconData[0] = 0x10; // frame type: url
+        beaconData[1] = (byte) 0xBA; // calibrated tx power at 0 m
+
+        builder.addServiceData(URI_BEACON_UUID, beaconData);
+
+        // Adding 0xFEAA to the "Service Complete List UUID 16" (0x3) for iOS compatibility
+        builder.addServiceUuid(URI_BEACON_UUID);
+
+        return builder.build();
+    }
+
+    // Build and return the ble advertising settings
+    private AdvertiseSettings getAdvertiseSettings() {
+        AdvertiseSettings.Builder builder = new AdvertiseSettings.Builder();
+        builder.setAdvertiseMode(AdvertiseSettings.ADVERTISE_MODE_LOW_POWER);
+        builder.setTxPowerLevel(AdvertiseSettings.ADVERTISE_TX_POWER_MEDIUM);
+        builder.setConnectable(false);
+
+        return builder.build();
+    }
+
+    // Surface a notification to the user that a URL is being broadcast
+    // The notification specifies the URL being broadcast (the long URL)
+    // and cannot be swiped away
+    private void createNotification() {
+                //.setPriority(NotificationCompat.PRIORITY_MIN);
+        Intent resultIntent = new Intent();
+        TaskStackBuilder stackBuilder = TaskStackBuilder.create(this);
+        stackBuilder.addParentStack(BroadcastActivity.class);
+        stackBuilder.addNextIntent(resultIntent);
+        PendingIntent resultPendingIntent = stackBuilder.getPendingIntent(0,
+            PendingIntent.FLAG_UPDATE_CURRENT);
+        registerReceiver(stopServiceReceiver, new IntentFilter("myFilter"));
+        PendingIntent pIntent = PendingIntent.getBroadcast(this, 0, new Intent("myFilter"),
+            PendingIntent.FLAG_UPDATE_CURRENT);
+        NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this)
+                .setSmallIcon(R.drawable.ic_leak_add_white_24dp)
+                .setContentTitle("Physical Web is broadcasting a URL")
+                .setContentText(displayUrl)
+                .setOngoing(true)
+                .addAction(2, "Stop", pIntent);
+        mBuilder.setContentIntent(resultPendingIntent);
+
+        NotificationManager mNotificationManager = (NotificationManager) getSystemService(
+            Context.NOTIFICATION_SERVICE);
+        mNotificationManager.notify(10, mBuilder.build());
+    }
+
+    protected BroadcastReceiver stopServiceReceiver = new BroadcastReceiver() {
```

Comment:
If this needs to be declared outside the method where it's used, move this up to the top and make it a member.  Use the m prefix.

However, I'm thinking it'd make the most sense just to create it in `createNotification`.